src/components/Sidebars/Payback.js

```diff
@@ -11,13 +19,19 @@ import Info from './shared/Info';
 import InfoContainer from './shared/InfoContainer';
 import lang from 'languages';
 import { MAX_UINT_BN } from '../../utils/units';
+import Loader from '../Loader';
 
 const Payback = ({ cdp, reset }) => {
   const { maker, newTxListener } = useMaker();
   const [amount, setAmount] = useState('');
   const [daiBalance, setDaiBalance] = useState(0);
   const [liquidationPrice, setLiquidationPrice] = useState(0);
   const [collateralizationRatio, setCollateralizationRatio] = useState(0);
+  const [hasAllowance, setHasAllowance] = useState(false);
```

Comment:
we might consider merging some of these `useStates` into a `useReducer`

Reply:
I've included in the latest commit a `useReducer` specific to allowance functionality. I haven't used this hook before now so be as liberal as you want with your critique! :slightly_smiling_face:

Reply:
you used it wonderfully 🙂